spec/models/weapon_spec.rb

```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+RSpec.describe Weapon, type: :model do
```

Comment:
I think there is a test missing, for the association for the `through` one, there is a shoulda matcher that may help...